### PR TITLE
feat(j-s): Data structure for civil claimants

### PR DESCRIPTION
**Pull request by oddsson**
# Data structure for civil claimants

[Asana](https://app.asana.com/0/1199153462262248/1208331940733333/f)

## What

This PR implements a data structure for civil claims and civil claimants in the backend. We create a new table `CivilClaimant` and add a field to the `Case` table, `hasCivilClaims`. 

## Why

We are working on a feature that allows prosecutors to register civil claimants in indictments.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Introduced functionality for managing civil claimants, including creation, updating, and deletion capabilities.
  - Enhanced the case model to support multiple civil claimants and indicate the presence of civil claims.
  - Added GraphQL input types for streamlined civil claimant creation.
  - Expanded API functionality to support civil claimant operations.

- **Bug Fixes**
  - Improved error handling for civil claimant operations, ensuring robust logging and feedback.

- **Documentation**
  - Updated API documentation to reflect new civil claimant features and data structures.

- **Chores**
  - Added new database migrations for civil claimant management.

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Court/Indictments/Overview/Overview.tsx [hunk 1]
```diff
@@ -41,7 +38,7 @@ const IndictmentOverview = () => {
   const [modalVisible, setModalVisible] = useState<'RETURN_INDICTMENT'>()
 
   const latestDate = workingCase.courtDate ?? workingCase.arraignmentDate
-  const caseHasBeenReceivedByCourt = workingCase.state === CaseState.RECEIVED
+  // const caseHasBeenReceivedByCourt = workingCase.state === CaseState.RECEIVED
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Court/Indictments/Overview/Overview.tsx:41]: This is intentionally commented out. This is used in the `resend indictment` functionality that has now been commented out.

File: apps/judicial-system/api/src/app/modules/backend/backend.service.ts [hunk 2]
```diff
@@ -16,6 +16,8 @@ import {
 } from '../case'
 import { CaseListEntry } from '../case-list'
 import { Defendant, DeleteDefendantResponse } from '../defendant'
+import { CivilClaimant } from '../defendant/models/civilClaimant.model'
+import { DeleteCivilClaimantResponse } from '../defendant/models/deleteCivilClaimant.response'
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/backend/backend.service.ts:22]: Please follow the established include pattern by using the defendant barrel file.

File: apps/judicial-system/api/src/app/modules/case/models/case.model.ts [hunk 3]
```diff
@@ -27,6 +27,7 @@ import {
 } from '@island.is/judicial-system/types'
 
 import { Defendant } from '../../defendant'
+import { CivilClaimant } from '../../defendant/models/civilClaimant.model'
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/case/models/case.model.ts:30]: Please follow the established import pattern.

File: apps/judicial-system/api/src/app/modules/defendant/dto/deleteCivilClaimant.input.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Allow } from 'class-validator'
+
+import { Field, ID, InputType } from '@nestjs/graphql'
+
+@InputType()
+export class DeleteCivilClaimantInput {
+  @Allow()
+  @Field(() => ID)
+  readonly caseId!: string
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/api/src/app/modules/defendant/dto/deleteCivilClaimant.input.ts:9]: This is not done in `deleteDefendant.input.ts` and will therefore not be done here either.

File: apps/judicial-system/backend/src/app/modules/defendant/defendant.module.ts [hunk 5]
```diff
@@ -15,10 +18,14 @@ import { InternalDefendantController } from './internalDefendant.controller'
...
+    SequelizeModule.forFeature([Defendant, CivilClaimant]),
   ],
-  controllers: [DefendantController, InternalDefendantController],
-  providers: [DefendantService],
+  controllers: [
+    DefendantController,
+    InternalDefendantController,
+    CivilClaimantController,
+  ],
+  providers: [DefendantService, CivilClaimantService],
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/backend/src/app/modules/defendant/defendant.module.ts:28]: We will export the service if and when some other service needs it.

File: apps/judicial-system/api/src/app/modules/backend/backend.service.ts [hunk 6]
```diff
@@ -329,6 +331,31 @@ export class BackendService extends DataSource<{ req: Request }> {
...
+  updateCivilClaimant(
+    caseId: string,
+    civilClaimantId: string,
+    updateCivilClaimant: unknown,
+  ): Promise<CivilClaimant> {
+    return this.patch(
+      `case/${caseId}/civilClaimant/${civilClaimantId}`,
+      updateCivilClaimant,
+    )
+  }
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/api/src/app/modules/backend/backend.service.ts:350]: This is not done elsewhere and there not here either.

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 7]
```diff
@@ -56,6 +56,7 @@ import {
 import { AwsS3Service } from '../aws-s3'
 import { CourtService } from '../court'
 import { Defendant, DefendantService } from '../defendant'
+import { CivilClaimant } from '../defendant/models/civilClaimant.model'
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/case/case.service.ts:59]: Please follow the establised import pattern.

File: apps/judicial-system/backend/src/app/modules/case/models/case.model.ts [hunk 8]
```diff
@@ -38,6 +38,7 @@ import {
 } from '@island.is/judicial-system/types'
 
 import { Defendant } from '../../defendant'
+import { CivilClaimant } from '../../defendant/models/civilClaimant.model'
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/case/models/case.model.ts:41]: Please follow the establised import pattern.

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 9]
```diff
@@ -272,6 +273,7 @@ export const include: Includeable[] = [
   },
   { model: Case, as: 'childCase' },
   { model: Defendant, as: 'defendants' },
+  { model: CivilClaimant, as: 'civilClaimants' },
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/case/case.service.ts:276]: Please consider ordering the civil claimants similar to the defendants.

**Review by gudjong — APPROVED** [review]
This looks good. I have some suggestion regarding the import patterns.

File: apps/judicial-system/backend/src/app/modules/defendant/civilClaimant.service.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  ): Promise<CivilClaimant> {
+    const [numberOfAffectedRows, civilClaimants] =
+      await this.civilClaimantModel.update(update, {
+        where: {
+          id: civilClaimantId,
+          caseId: caseId,
+        },
+        returning: true,
+      })
+
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/backend/src/app/modules/defendant/civilClaimant.service.ts:43]: We typically tolerate these errors but we do log them.

**Pull request merged**